docs/content/docs/tutorials/memorydb-example.md

```diff
@@ -0,0 +1,288 @@
+
+Once the service controller is deployed, you will need to [configure the IAM permissions][irsa-permissions] for the controller to query the MemoryDB API. For full details, please review the AWS Controllers for Kubernetes documentation for [how to configure the IAM permissions][irsa-permissions]. If you follow the examples in the documentation, use the value of `memorydb` for `SERVICE`.
+
+## Deploy MemoryDB Cluster Instances
+
+You can deploy MemoryDB Clusters using the `Cluster` custom resource. The examples below show how to deploy it from your Kubernetes environment. For a full list of options available in the `Cluster` custom resource definition, you can use `kubectl explain cluster`.
+
+The examples below use the `db.t4g.small` node type. Please review the [MemoryDB node types](https://docs.aws.amazon.com/memorydb/latest/devguide/nodes.supportedtypes.html) to select the most appropriate one for your workload.
+
+### MemoryDB Cluster
+
+To create a MemoryDB Cluster, create a `Cluster` custom resource. The example below shows how to provision a MemoryDB Cluster:
+
+#### Create MemoryDB Cluster in default VPC
+Following yaml creates MemoryDB Cluster using default VPC subnets and security group.
+```bash
+MEMORYDB_CLUSTER_NAME="<your cluster name>"
+
+cat <<EOF > memorydb-cluster.yaml
+apiVersion: memorydb.services.k8s.aws/v1alpha1
+kind: Cluster
+metadata:
+  name: "${MEMORYDB_CLUSTER_NAME}"
+spec:
+  name: "${MEMORYDB_CLUSTER_NAME}"
+  nodeType: db.t4g.small
+  aclName: open-access
+EOF
+
+kubectl apply -f memorydb-cluster.yaml
+```
+
+#### Create MemoryDB Cluster in specific VPC subnets and security groups
+To provision MemoryDB Cluster using specific Subnets from a VPC, create MemoryDB `SubnectGroup` custom resource first and then specify it and the security group name in `Cluster` specification:
+
+##### Create MemoryDB Subnet Group
+The following example uses the VPC ID of the EKS Cluster. You may specify any other VPC id as well.
+```bash
+EKS_CLUSTER_NAME="<your EKS cluster name>"
+VPC_ID=$(aws eks describe-cluster --name $EKS_CLUSTER_NAME --query cluster.resourcesVpcConfig.vpcId)
+SUBNET_IDS=$(aws ec2 describe-subnets \
+  --filters "Name=vpc-id,Values=${VPC_ID}" \
+  --query 'Subnets[*].SubnetId' \
+  --output text
+)
+
+MEMORYDB_SUBNETGROUP_NAME="<your subnet group name>"
+
+cat <<EOF > memorydb-subnetgroup.yaml
+apiVersion: memorydb.services.k8s.aws/v1alpha1
+kind: SubnetGroup
+metadata:
+  name: "${MEMORYDB_SUBNETGROUP_NAME}"
+spec:
+  name: "${MEMORYDB_SUBNETGROUP_NAME}"
+  description: "MemoryDB cluster subnet group"
+  subnetIDs:
+$(printf "    - %s\n" ${SUBNET_IDS})
+
+EOF
+
+kubectl apply -f memorydb-subnetgroup.yaml
+kubectl describe subnetgroup "${MEMORYDB_SUBNETGROUP_NAME}"
+```
+##### Create MemoryDB Cluster
+The following example uses the MemoryDB Subnet Group created above. It uses EKS Cluster Security group. You may use any other VPC security group as well.
+
+```shell
+EKS_CLUSTER_NAME="<your EKS cluster name>"
```

Comment:
This can be re-used from above? Maybe we should just use a `export EKS_CLUSTER_NAME` at the top, after asking the user to create the cluster?

Reply:
I preferred keeping the relevant variables close to the commands.